### PR TITLE
Progress table: keep expanded rows between detail/summary views

### PR DESCRIPTION
**Pull request by maureensturgeon**
- Removes `ProgressTableDetailView` and `ProgressTableSummaryView` and replace with modified `ProgressTableContainer`
  - Updates associated tests and storybooks
- Detail/summary cell renderer functions added to `progressTableHelpers`

https://user-images.githubusercontent.com/24235215/116283589-e5cb8d00-a740-11eb-86e4-99f4bce908ca.mov

## Testing story
Added test for helper functions, moved all summary and detail view-specific tests into ProgressTableContainer tests. Also tested locally.

## Follow-up work
Rename `ProgressTableContainer` to `ProgressTableView` (I'm not doing this now so that the code changes are somewhat easier to review)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx]: to me this name creates some ambiguity around whether this will return the number of extra rows per student, or the total number of extra rows currently in the state. i realize that `numDetailRows` isn't any better though. what do you think of `numDetailRowsPerStudent`?

Comment (cforkish) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx]: oh haha if i read the implementation it's pretty obvious given `const rowsPerStudent`... but i think it would still be helpful to surface that in the name

**Review by cforkish — COMMENTED** [review]
feel free to consider this suggestion out of scope, but this would be a great time to update our internal naming from `summary/detail` to `lesson/level`. externally we're referring to the expansion rows as "progress details", so it would be great to disambiguate "detail rows" from "level cells". but that would could easily be deferred until the larger renaming task we have planned.

File: apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js]: NIT: imo we should use block-style comments for functions, classes, etc.

that said, good work documenting this functionality! :)

Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js]: Yes! I actually updated this locally just haven't pushed it up yet!

Comment (cforkish) [reply apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js]: 🙌

**Review by cforkish — APPROVED** [review]
a few minor comments, but nothing blocking. overall this looks great!

Screen Shot 2021-04-26 at 12 20 28 PM🎉

File: apps/src/templates/sectionProgress/SectionProgress.jsx
Comment (cforkish) [inline apps/src/templates/sectionProgress/SectionProgress.jsx]: don't we need to make sure `currentView !== ViewType.STANDARDS`? or return null from container's render in that case?

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:290]: NIT: a comment that this is the container for the table+legend would be helpful. not worth adding unless making other changes though

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.story.jsx
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.story.jsx:97]: NIT: is it possible to nest storybooks so we could keep the summary/detail stories in separate lists rather than making the names super long?

File: apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js:22]: nice encapsulation here. seems like a pretty clean solution to a pretty dirty problem. :)

**Review by cforkish — APPROVED** [review]
LGTM! please confirm that the standards view is properly handled before merging though.

**Review by jamescodeorg — APPROVED** [review]
Nice improvements!  Let's add ProgressTableContainer --> ProgressTable to our list of pending renames.

File: apps/src/templates/sectionProgress/SectionProgress.jsx
Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/SectionProgress.jsx]: 😱   I was using a CSD course and didn't take into account the standards view! Excellent catch! I will update and add a test

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.story.jsx
Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.story.jsx:97]: This wasn't super straight forward without making separate story files, so I'm going to hold off for now.

**Pull request merged**
